project/Build.scala

```diff
@@ -56,6 +56,16 @@ object Build extends sbt.Build {
   val newScalaBinaryVersionsInThisRelease: Set[String] =
     Set()
 
+  val javaVersion: Int = {
+    val v = System.getProperty("java.version")
+    v.substring(0, 3) match {
+      case "1.8" => 8
+      case "1.7" => 7
+      case "1.6" => 6
+      case _     => sys.error(s"Unknown JDK version $v")
+    }
```

Comment:
If we want to assert in our CI that we know the JDK version we're testing, we can add a *task* `ensureKnownJavaVersion` that will thrown if `javaVersion.isEmpty`, and we can call that task from the matrix.